app/Bots/IRCAnnounceBotExternal.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * NOTICE OF LICENSE.
+ *
+ * UNIT3D Community Edition is open-sourced software licensed under the GNU Affero General Public License v3.0
+ * The details is bundled with this project in the file LICENSE.txt.
+ *
+ * @project    UNIT3D Community Edition
+ *
+ * @author     HDVinnie <hdinnovations@protonmail.com>
+ * @license    https://www.gnu.org/licenses/agpl-3.0.en.html/ GNU Affero General Public License v3.0
+ */
+
+namespace App\Bots;
+
+use App\Models\TmdbMovie;
+use App\Models\TmdbTv;
+use App\Models\Torrent;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Throwable;
+
+class IRCAnnounceBotExternal
+{
+    public static function postAnnounceMsg(Torrent $torrent): bool
+    {
+        if (! config('irc-bot-external.is_enabled')) {
+            return false;
+        }
+
+        $appurl = config('app.url');
+        $announceTypeEnum = 0; // 0 NEW
+
+        $originEnum = 0;
+
+        if ($torrent->internal) {
+            $originEnum = 1;
+        }
+
+        if ($torrent->personal_release) {
+            $originEnum = 2;
+        }
+
+        $leechTypeEnum = 0;
+
+        if ($torrent->free > 0) {
+            switch ($torrent->free) {
+                case 100:
+                    $leechTypeEnum = 1;
+
+                    break;
+                case 75:
+                    $leechTypeEnum = 2;
+
+                    break;
+                case 50:
+                    $leechTypeEnum = 3;
+
+                    break;
+                case 25:
+                    $leechTypeEnum = 4;
+
+                    break;
+            }
+        }
+
+        if ($torrent->doubleup) {
+            switch ($leechTypeEnum) {
+                case 0:
+                    $leechTypeEnum = 5;
+
+                    break;
+                case 1:
+                    $leechTypeEnum = 6;
+
+                    break;
+                case 2:
+                    $leechTypeEnum = 7;
+
+                    break;
+                case 3:
+                    $leechTypeEnum = 8;
+
+                    break;
+                case 4:
+                    $leechTypeEnum = 9;
+            }
+        }
+
+        $meta = null;
+        $category = $torrent->category;
+
+        if ($torrent->tmdb_movie_id > 0 || $torrent->tmdb_tv_id > 0) {
+            $meta = match (true) {
+                $category->tv_meta    => TmdbTv::find($torrent->tmdb_tv_id),
+                $category->movie_meta => TmdbMovie::find($torrent->tmdb_movie_id),
+                default               => null,
+            };
+        }
+
+        return self::post([
+            'id'                     => (string) $torrent->id,
```

Comment:
Just want to double check that you intend for this to be a string instead of an int.

Reply:
If it will ever only be used with UNIT3D then int is fine but if someone have string IDs then that won't be ideal. Probably better I handle that on the other end and we send as int here :+1:

Reply:
If you want maximum flexibility for in the very very rare chance we change to uuids in the future, then keep the string :). I just wanted to make sure you weren't copying the api which does in fact use a string for this if you didn't intend to match it.